editor/editor_inspector.cpp

```diff
@@ -4337,17 +4338,37 @@ void EditorInspector::_set_property_favorited(const String &p_path, bool p_favor
 		return;
 	}
 
-	StringName class_name = object->get_class_name();
-	while (!class_name.is_empty()) {
+	StringName validate_name = object->get_class_name();
+	StringName class_name;
+	// Check if the property is built-in.
+	while (!validate_name.is_empty()) {
+		class_name = validate_name;
+
 		bool has_prop = ClassDB::has_property(class_name, p_path, true);
 		if (has_prop) {
 			break;
 		}
 
-		class_name = ClassDB::get_parent_class_nocheck(class_name);
+		validate_name = ClassDB::get_parent_class_nocheck(class_name);
+	}
+
+	if (validate_name.is_empty()) {
+		// "script" isn't a real property, so a hack is necessary.
+		if (p_path != "script") {
+			class_name = validate_name;
+		}
+
+		// Deal with theme properties.
+		if (p_path.begins_with("theme_override_")) {
+			const String slice = p_path.get_slice("/", 1);
+			if (!slice.is_empty()) {
+				class_name = validate_name;
+			}
+		}
```

Comment:
This doesn't seem to work, overrides still print errors.

Reply:
Try again.

Reply:
There is no error now, but the property gets unfavorited after you re-select the node.

Reply:
Third time is the charm.